### PR TITLE
add inline checkstyle

### PR DESCRIPTION
**Pull request by jetersen**
this should improve the situation, we don't have to move files around.
Now the config is centralized here in our parent pom 🚀

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -96,6 +96,17 @@
...
           </executions>
         </plugin>
+        <plugin>
+          <artifactId>maven-checkstyle-plugin</artifactId>
+          <version>3.1.1</version>
+          <dependencies>
+            <dependency>
+              <groupId>com.puppycrawl.tools</groupId>
+              <artifactId>checkstyle</artifactId>
+              <version>8.36.2</version>
```
Comment (jetersen) [inline pom.xml:106]:
```suggestion
              <version>8.38</version>
```

**Review by JesperMonsted — APPROVED** [review]
Look cool, we can always add more checks along the way.

File: pom.xml [hunk 2]
```diff
@@ -157,6 +168,49 @@
...
+          <includeTestSourceDirectory>true</includeTestSourceDirectory>
+          <sourceDirectories>
+            <sourceDirectory>${project.build.sourceDirectory}</sourceDirectory>
+          </sourceDirectories>
+          <testSourceDirectories>
+            <sourceDirectory>${project.build.testSourceDirectory}</sourceDirectory>
+          </testSourceDirectories>
+          <checkstyleRules>
+            <module name="Checker">
+              <module name="FileTabCharacter" />
```
Comment (jetersen) [inline pom.xml:183]: This ensures spaces is used instead of tabs as whitespace.

File: pom.xml [hunk 3]
```diff
@@ -157,6 +168,49 @@
...
+              <module name="NewlineAtEndOfFile">
+                <property name="lineSeparator" value="lf" />
+              </module>
```
Comment (jetersen) [inline pom.xml:186]: This ensures line separator is lf (ie. Linux line endings)

File: pom.xml [hunk 4]
```diff
@@ -157,6 +168,49 @@
...
+              <module name="RegexpSingleline">
+                <property name="format" value="\s+$"/>
+                <property name="message" value="Trailing spaces are not allowed."/>
+              </module>
```
Comment (jetersen) [inline pom.xml:190]: This checks for any whitespace.

File: pom.xml [hunk 5]
```diff
@@ -157,6 +168,49 @@
...
+              <module name="TreeWalker">
+                <module name="ImportOrder">
+                  <property name="separatedStaticGroups" value="true" />
+                  <property name="option" value="bottom" />
+                </module>
```
Comment (jetersen) [inline pom.xml:195]: This validates import order A-Z and static is grouped separately.

File: pom.xml [hunk 6]
```diff
@@ -157,6 +168,49 @@
...
+                <module name="UnusedImports" />
```
Comment (jetersen) [inline pom.xml:196]: This checks if we have unused dependencies.

File: pom.xml [hunk 7]
```diff
@@ -157,6 +168,49 @@
...
+                <module name="AvoidStarImport" />
```
Comment (jetersen) [inline pom.xml:197]: This checks if we have any start imports.

File: pom.xml [hunk 8]
```diff
@@ -157,6 +168,49 @@
...
+              </module>
+              <module name="SuppressionSingleFilter">
+                <property name="checks" value=".+"/>
+                <property name="files" value=".+\.properties"/>
+              </module>
```
Comment (jetersen) [inline pom.xml:202]: This excludes properties file for the above checkstyles.

**Comment by jetersen** [comment]
I think it would be best to configure checkstyle to not fail the mvn execution step and use https://github.com/jwgmeligmeyling/checkstyle-github-action as the way to annotate the build.

**Review by JesperMonsted — APPROVED** [review]

**Pull request merged**